From 1a9953c39a08407607e5ef9d2ca4fefa77b5eaea Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Fri, 12 Apr 2024 11:33:08 +0800
Subject: [PATCH] wifi: mt76: mt7915: fix rx filter setting for bfee
 functionality

Fix rx filter setting to prevent dropping NDPA frames. Without this
commit, bfee functionality may behave abnormally.

Fixes: e57b7901469f ("mt76: add mac80211 driver for MT7915 PCIe-based chipsets")
Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
---
 mt7915/main.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mt7915/main.c
+++ b/mt7915/main.c
@@ -564,8 +564,7 @@ static void mt7915_configure_filter(stru
 
 	MT76_FILTER(CONTROL, MT_WF_RFCR_DROP_CTS |
 			     MT_WF_RFCR_DROP_RTS |
-			     MT_WF_RFCR_DROP_CTL_RSV |
-			     MT_WF_RFCR_DROP_NDPA);
+			     MT_WF_RFCR_DROP_CTL_RSV);
 
 	*total_flags = flags;
 	rxfilter = phy->rxfilter;
